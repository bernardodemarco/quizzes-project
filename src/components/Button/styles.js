import styled, { css } from 'styled-components';

export const Button = styled.button`
  ${({ theme, background, fontSize, fontWeight, lineHeight }) => css`
    background-color: ${background ? theme.colors.primary : theme.colors.white};
    color: ${background ? theme.colors.white : theme.colors.dark};
    font-size: ${theme.fonts.fontSizes.buttons[fontSize]};
    font-weight: ${theme.fonts.fontWeights[fontWeight]};
    line-height: ${lineHeight};

    font-family: inherit;
    width: 100%;
    height: 5.6rem;
    text-align: center;
    border: none;
    border-radius: 1.6rem;
    padding: 1.6rem 3.2rem;

    &:hover {
      cursor: pointer;
    }
  `}
`;
