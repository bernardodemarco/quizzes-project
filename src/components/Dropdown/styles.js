import styled, { css } from 'styled-components';

export const Dropdown = styled.ul`
  ${({ theme, isVisible }) => css`
    visibility: hidden;
    opacity: 0;

    transition: all 300ms ease-in-out;

    ${
      isVisible &&
      css`
      visibility: visible;
      opacity: 1;
    `
    }

    position: absolute;
    top: 10rem;
    right: 38%;

    width: 14.5rem;
    height: 15.2rem;

    background-color: ${theme.colors.white};
    border-radius: 0.8rem;
    box-shadow: 0rem 0.4rem 0.4rem rgba(0, 0, 0, 0.25);

    list-style: none;

    @media (max-width: 785px) {
      visibility: hidden;
      opacity: 0;
    }
  `}
`;

export const DropdownItem = styled.li`
  ${({ theme }) => css`
    width: 100%;

    padding: 1rem 4.5rem;

    border-bottom: 0.1rem solid ${theme.colors.gray};

    &:last-child {
      border-bottom: none;
    }

    &:hover {
      cursor: pointer;
    }
  `}
`;

export const Text = styled.p`
  ${({ theme }) => css`
    color: #000000;

    text-align: center;

    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    font-weight: ${theme.fonts.fontWeights.normal};
    line-height: 1.8rem;
  `}
`;
