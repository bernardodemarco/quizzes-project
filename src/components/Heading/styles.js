import styled, { css } from 'styled-components';

export const Heading = styled.h1`
  ${({ theme, fontSize, fontWeight, lineHeight }) => css`
    color: ${theme.colors.dark};
    font-size: ${theme.fonts.fontSizes.headings[fontSize]};
    font-weight: ${theme.fonts.fontWeights[fontWeight]};
    line-height: ${lineHeight};
  `}
`;
