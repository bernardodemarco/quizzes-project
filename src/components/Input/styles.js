import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    border: 0.1rem solid ${theme.colors.gray};
    border-radius: 1.2rem;

    width: 100%;
    height: 5.3rem;
    padding: 1.6rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
  `}
`;

export const Input = styled.input`
  ${({ theme }) => css`
    width: 75%;
    border: none;
    font-family: inherit;
    font-weight: ${theme.fonts.fontWeights.light};
    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    color: ${theme.colors.darkGray};
  `}
`;

export const Icon = styled.img`
  ${({ hover = false }) => css`
    width: 3rem;
    height: 2.5rem;

    &:hover {
      cursor: ${hover ? 'pointer' : 'default'};
    }
  `}
`;
