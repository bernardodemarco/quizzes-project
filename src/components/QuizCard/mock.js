export const answeredQuiz = {
  title: 'UI',
  shortDescription: 'Um quiz sobre UI que vai mudar sua vida!',
  imgSrc:
    'https://sujeitoprogramador.com/wp-content/uploads/2020/12/uiDesign-1.png',
  numOfQuestions: 3,
  difficulty: 'easy',
  isAnswered: true,
  answeredDate: '2022-06-21 22:32',
  numOfCorrectAnswers: 1,
};

export const notAnsweredQuiz = {
  title: 'Conceitos b√°sicos de HTML',
  shortDescription: 'Um quiz sobre HTML que vai mudar sua vida!',
  date: '2022-06-19 14:32',
  imgSrc:
    'https://www.hostinger.com.br/tutoriais/wp-content/uploads/sites/12/2018/07/Como-inserir-imagem-ou-logo-usando-HTML-no-seu-site-1.png',
  numOfQuestions: 4,
  difficulty: 'easy',
  isAnswered: false,
  answeredDate: null,
  numOfCorrectAnswers: null,
};
