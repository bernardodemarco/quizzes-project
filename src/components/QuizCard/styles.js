import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    overflow: hidden;

    border: 0.1rem solid ${theme.colors.gray};
    border-radius: 0.8rem;
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));

    width: 55.2rem;
    height: fit-content;
    background-color: ${theme.colors.white};

    &:hover {
      cursor: pointer;
    }

    @media ${theme.media.largeDevices} {
      width: 45rem;
    }

    @media ${theme.media.mediumDevices} {
      width: 55.2rem;
    }

    @media ${theme.media.smallDevices} {
      width: 90vw;

      p {
        text-align: start;
      }
    }
  `}
`;

export const Image = styled.div`
  ${({ imgSrc }) => css`
    // position set to relative in order to position <QuizDifficulty /> accordingly
    position: relative;

    width: 100%;
    height: 19.4rem;

    background-image: url(${imgSrc});
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
  `}
`;

export const TextContainer = styled.div`
  ${() => css`
    width: 100%;
    padding: 1.6rem 1.6rem 0.8rem 1.6rem;

    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 0.4rem;
  `}
`;

export const FeedbackContainer = styled.div`
  ${() => css`
    width: 100%;

    display: flex;
    justify-content: space-between;
  `}
`;
