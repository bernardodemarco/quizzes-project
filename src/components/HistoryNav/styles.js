import styled, { css } from 'styled-components';

import { Container as PageContainer } from './../PageContainer/styles';
import { ImageContainer as ReturnButton } from './../ReturnButton/styles';
import { Heading } from './../Heading/styles';

export const Nav = styled.nav`
  ${({ theme }) => css`
    z-index: 1;
    position: fixed;
    top: 0;
    right: 0;
    left: 0;

    height: 12rem;
    border-bottom: 0.1rem solid ${theme.colors.gray};

    ${PageContainer} {
      min-height: initial;
      height: 100%;
      justify-content: flex-start;
      align-items: center;
    }

    ${Heading} {
      z-index: 1;
      position: fixed;
      left: 14.7rem;
    }

    @media ${theme.media.mediumDevices} {
      ${ReturnButton} {
        left: 5rem;
      }
    }

    @media ${theme.media.smallDevices} {
      ${Heading} {
        left: 12rem;
        font-size: ${theme.fonts.fontSizes.headings.medium};
      }
    }
  `}
`;
