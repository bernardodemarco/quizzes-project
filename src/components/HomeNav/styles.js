import styled, { css } from 'styled-components';

import { Container as PageContainer } from '../PageContainer/styles';
import { Input, Container as InputContainer } from '../Input/styles';
import { Heading } from '../Heading/styles';

export const Button = styled.button`
  ${({ theme }) => css`
    display: none;

    z-index: 2;
    position: fixed;
    top: 3rem;
    right: 3rem;

    height: 4rem;
    width: 4rem;

    color: ${theme.colors.primary};
    background-color: ${theme.colors.lightGray};
    border: none;

    &:hover {
      cursor: pointer;
    }

    @media (max-width: 785px) {
      display: block;
    }
  `}
`;

export const Container = styled.div`
  ${({ theme, isVisible }) => css`
    height: 12rem;

    z-index: 1;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;

    border-bottom: 0.1rem solid ${theme.colors.gray};

    transition: all 300ms ease-in-out;

    & ${PageContainer} {
      min-height: initial;
      height: 100%;
    }

    @media (max-width: 1300px) {
      & ${PageContainer} {
        padding: 0 3rem;
      }
    }

    @media (max-width: 785px) {
      & {
        height: 100vh;
        bottom: 0;
        border-bottom: none;

        visibility: hidden;
        opacity: 0;
        ${
          isVisible &&
          css`
          visibility: visible;
          opacity: 1;
        `
        }
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${() => css`
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-end;

    @media (max-width: 785px) {
      & {
        height: 100%;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        gap: 7.5rem;
      }
    }
  `}
`;

export const Logo = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    align-items: flex-start;

    & ${Heading} {
      text-align: center;
    }

    @media (max-width: 1130px) {
      & ${Heading} {
        font-size: ${theme.fonts.fontSizes.headings.medium};
        margin-right: 2rem;
      }
    }

    @media ${theme.media.mediumDevices} {
      & ${Heading} {
        font-size: ${theme.fonts.fontSizes.headings.small};
      }
    }

    @media (max-width: 785px) {
      & {
        justify-content: center;
        align-items: center;
      }

      & ${Heading} {
        margin-right: 0;
      }
    }
  `}
`;

export const Nav = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    gap: 3rem;

    & ${Input} {
      width: 40.4rem;
    }

    @media (max-width: 1130px) {
      & {
        align-items: center;
      }

      & ${Input} {
        width: 30rem;
      }
    }

    @media ${theme.media.mediumDevices} {
      & ${Input} {
        width: 22.5rem;
      }
    }

    @media (max-width: 785px) {
      & {
        flex-direction: column;
        justify-content: center;
        align-items: center;
      }

      & ${InputContainer}, & ${Input}, & .dropdown-button  {
        display: none;
      }
    }
  `}
`;

export const Link = styled.a`
  ${({ theme }) => css`
    display: block;
    text-align: center;
    color: ${theme.colors.dark};

    font-weight: ${theme.fonts.fontWeights.light};
    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    line-height: 2.1rem;

    &:hover {
      cursor: pointer;
    }

    @media (max-width: 785px) {
      & {
        font-size: 1.8rem;
      }
    }
  `}
`;

export const Theme = styled.p`
  ${({ theme }) => css`
    display: none;
    color: ${theme.colors.dark};
    text-align: center;

    font-weight: ${theme.fonts.fontWeights.light};
    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    line-height: 2.1rem;

    &:hover {
      cursor: pointer;
    }

    @media (max-width: 785px) {
      & {
        display: block;
        font-size: 1.8rem;
      }
    }
  `}
`;
