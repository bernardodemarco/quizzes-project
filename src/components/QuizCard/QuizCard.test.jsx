import { fireEvent, screen } from '@testing-library/react';

import { QuizCard } from '.';
import { answeredQuiz, notAnsweredQuiz } from './mock';

import { theme } from '../../styles/theme';
import { renderThemeProvider } from './../../styles/render-theme-provider';

describe('<QuizCard />', () => {
  it('should render an answered <QuizCard /> correctly', () => {
    const testFunc = jest.fn();
    const { container } = renderThemeProvider(
      <QuizCard onClick={testFunc} {...answeredQuiz} />,
    );

    expect(
      screen.getByRole('heading', { name: answeredQuiz.title }),
    ).toBeInTheDocument();

    expect(screen.getByText(answeredQuiz.shortDescription)).toBeInTheDocument();

    expect(
      screen.getByText(`Em ${answeredQuiz.answeredDate}`),
    ).toBeInTheDocument();

    expect(
      screen.getByText(
        `Você acertou ${answeredQuiz.numOfCorrectAnswers} de ${answeredQuiz.numOfQuestions}`,
      ),
    ).toHaveStyle({
      color: theme.colors.error,
    });

    expect(screen.queryByText(answeredQuiz.difficulty)).not.toBeInTheDocument();

    expect(container.firstChild.firstChild).toHaveStyle({
      'background-image': `url(${answeredQuiz.imgSrc})`,
    });

    fireEvent.click(container.firstChild);
    expect(testFunc).toHaveBeenCalledTimes(1);
  });

  it('should render quiz feedback with the correct colors', () => {
    const testFunc = jest.fn();
    renderThemeProvider(
      <QuizCard
        onClick={testFunc}
        {...answeredQuiz}
        numOfCorrectAnswers={answeredQuiz.numOfQuestions}
      />,
    );
    expect(
      screen.getByText(
        `Você acertou ${answeredQuiz.numOfQuestions} de ${answeredQuiz.numOfQuestions}`,
      ),
    ).toHaveStyle({
      color: theme.colors.success,
    });
  });

  it('should render an unanswered <QuizCard /> correctly', () => {
    const testFunc = jest.fn();
    const { container } = renderThemeProvider(
      <QuizCard onClick={testFunc} {...notAnsweredQuiz} />,
    );

    expect(
      screen.getByRole('heading', { name: notAnsweredQuiz.title }),
    ).toBeInTheDocument();

    expect(
      screen.getByText(notAnsweredQuiz.shortDescription),
    ).toBeInTheDocument();

    expect(screen.getByText(notAnsweredQuiz.difficulty)).toHaveStyle({
      'background-color': theme.colors.secondary,
    });

    expect(
      screen.queryByText(`Em ${notAnsweredQuiz.answeredDate}`),
    ).not.toBeInTheDocument();

    expect(
      screen.queryByText(
        `Você acertou ${notAnsweredQuiz.numOfCorrectAnswers} de ${answeredQuiz.numOfQuestions}`,
      ),
    ).not.toBeInTheDocument();

    expect(container.firstChild.firstChild).toHaveStyle({
      'background-image': `url(${notAnsweredQuiz.imgSrc})`,
    });

    fireEvent.click(container.firstChild);
    expect(testFunc).toHaveBeenCalledTimes(1);
  });

  it('should render <QuizDifficulty /> with the right background-colors', () => {
    const testFunc = jest.fn();
    renderThemeProvider(
      <QuizCard onClick={testFunc} {...notAnsweredQuiz} difficulty="hard" />,
    );

    expect(screen.getByText('hard')).toHaveStyle({
      'background-color': theme.colors.error,
    });
  });

  it('should match snapshot', () => {
    const testFunc = jest.fn();
    const { container } = renderThemeProvider(
      <QuizCard onClick={testFunc} {...answeredQuiz} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
