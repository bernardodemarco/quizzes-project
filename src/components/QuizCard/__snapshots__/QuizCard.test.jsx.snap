// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuizCard /> should match snapshot 1`] = `
.c0 {
  overflow: hidden;
  border: 0.1rem solid #BEBAB3;
  border-radius: 0.8rem;
  -webkit-filter: drop-shadow(0px 4px 4px rgba(0,0,0,0.25));
  filter: drop-shadow(0px 4px 4px rgba(0,0,0,0.25));
  width: 55.2rem;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  background-color: #FFFFFF;
}

.c0:hover {
  cursor: default;
}

.c1 {
  position: relative;
  width: 100%;
  height: 19.4rem;
  background-image: url(https://sujeitoprogramador.com/wp-content/uploads/2020/12/uiDesign-1.png);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

.c2 {
  width: 100%;
  padding: 1.6rem 1.6rem 0.8rem 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 0.4rem;
}

.c3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  font-size: 1.2rem;
  font-weight: 500;
  color: #EF4949;
  text-align: center;
  line-height: 1.8rem;
}

.c5 {
  font-size: 1.2rem;
  font-weight: 500;
  color: #78746D;
  text-align: center;
  line-height: 1.8rem;
}

.c7 {
  font-size: 1.4rem;
  font-weight: 400;
  color: #3C3A36;
  text-align: center;
  line-height: 2.1rem;
}

.c6 {
  color: #3C3A36;
  font-size: 2.4rem;
  font-weight: 500;
  line-height: normal;
}

@media (max-width:1200px) {
  .c0 {
    width: 45rem;
  }
}

@media (max-width:974px) {
  .c0 {
    width: 55.2rem;
  }
}

@media (max-width:575px) {
  .c0 {
    width: 90vw;
  }

  .c0 p {
    text-align: start;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <p
        class="c4"
        color="error"
        font-size="small"
        font-weight="normal"
      >
        VocÃª acertou 1 de 3
      </p>
      <p
        class="c5"
        color="darkGray"
        font-size="small"
        font-weight="normal"
      >
        Em 2022-06-21 22:32
      </p>
    </div>
    <h3
      class="c6"
      font-size="small"
      font-weight="normal"
    >
      UI
    </h3>
    <p
      class="c7"
      color="dark"
      font-size="medium"
      font-weight="light"
    >
      Um quiz sobre UI que vai mudar sua vida!
    </p>
  </div>
</div>
`;
